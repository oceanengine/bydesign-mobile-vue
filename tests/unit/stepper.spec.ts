import { mount, later } from '../index';
import { Stepper as NativeStepper } from '@src';
const Stepper = NativeStepper as any;
later;
describe('Stepper', () => {
  test('disabled stepper', () => {
    const wrapper = mount(Stepper, {
      propsData: {
        disabled: true,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
  test('click event', async () => {
    const wrapper = mount(Stepper, {
      propsData: {
        value: 1,
        max: 3,
      },
    }) as any;

    const plus = wrapper.find('.byted-stepper__increase');
    const minus = wrapper.find('.byted-stepper__decrease');
    plus.trigger('click');
    plus.trigger('click');
    minus.trigger('click');
    minus.trigger('click');
    await later(100);

    expect(wrapper.emitted('input')).toEqual([[2], [2], [0], [0]]);
    expect(wrapper.vm.value).toBe(1);
  });
  test('min value is 0', () => {
    const wrapper = mount(Stepper, {
      propsData: {
        value: 1,
        min: 0,
      },
    }) as any;

    const input = wrapper.find('input');
    input.element.value = '';
    input.trigger('input');
    input.trigger('blur');

    expect(input.element.value).toEqual('');
  });

  test('step prop', async () => {
    const wrapper = mount({
      template: `
        <byted-stepper v-model="value" :min="1" :max="100" :step="10" />
      `,
      components: {
        'byted-stepper': Stepper,
      },
      data() {
        return {
          value: 0,
        };
      },
    }) as any;
    await later();
    const plus = wrapper.find('.byted-stepper__increase');
    await plus.trigger('click');
    expect(wrapper.vm.value).toEqual(10);
  });

  test('change event', async () => {
    const onChange = jest.fn();
    const wrapper = mount(Stepper, {
      propsData: {
        value: 2,
      },
      listeners: {
        change: onChange,
      },
    }) as any;

    const plus = wrapper.find('.byted-stepper__increase');
    await plus.trigger('click');
    expect(onChange).toBeCalledTimes(1);
  });
});
