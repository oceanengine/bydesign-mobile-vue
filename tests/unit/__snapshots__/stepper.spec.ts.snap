// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper disabled stepper 1`] = `
<div class="byted-stepper">
  <div class="byted-stepper__decrease byted-stepper__crease byted-stepper__crease--normal byted-stepper__crease--disabled"><svg viewBox="0 0 48 48" xmlns="http://www.w3.org/2000/svg" class="byted-stepper__desvg byted-stepper__desvg--normal byted-stepper__disable">
      <path fill-rule="evenodd" clip-rule="evenodd" d="M8.5 24c0-1.1.9-2 2-2h28a2 2 0 110 4h-28a2 2 0 01-2-2z"></path>
    </svg></div>
  <div class="byted-stepper__number byted-stepper__number--normal byted-stepper__number--disabled"><input type="number" class="byted-stepper__input">1</div>
  <div class="byted-stepper__increase byted-stepper__crease byted-stepper__crease--normal byted-stepper__crease--disabled"><svg viewBox="0 0 48 48" xmlns="http://www.w3.org/2000/svg" class="byted-stepper__insvg byted-stepper__insvg--normal byted-stepper__disable">
      <path fill-rule="evenodd" clip-rule="evenodd" d="M26.06 10a2 2 0 00-4 0l-.01 12H10a2 2 0 100 4h12.04l-.02 12a2 2 0 004 0l.02-12H38a2 2 0 100-4H26.05l.01-12z"></path>
    </svg></div>
</div>
`;
